linode/data_source_linode_images_test.go

```diff
@@ -0,0 +1,100 @@
+package linode
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"fmt"
+	"strconv"
+	"testing"
+)
+
+func TestAccDataSourceLinodeImages_basic(t *testing.T) {
+	t.Parallel()
+
+	imageID := "linode/debian8"
+	resourceName := "data.linode_images.foobar"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceLinodeImagesBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "image.0.id", imageID),
+					resource.TestCheckResourceAttr(resourceName, "image.0.label", "Debian 8"),
+					resource.TestCheckResourceAttr(resourceName, "image.0.description", ""),
+					resource.TestCheckResourceAttr(resourceName, "image.0.is_public", "true"),
+					resource.TestCheckResourceAttr(resourceName, "image.0.type", "manual"),
+					resource.TestCheckResourceAttr(resourceName, "image.0.size", "1300"),
+					resource.TestCheckResourceAttr(resourceName, "image.0.vendor", "Debian"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceLinodeImages_noFilters(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "data.linode_images.foobar"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceLinodeImagesNoFilters(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceCheckLinodeAllImagesNotEmpty(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceCheckLinodeAllImagesNotEmpty(resourceName string) resource.TestCheckFunc {
```

Comment:
I recognize that this is a common thing, could this be a more abstract helper? For instance in the test helpers file you could make the function `testAccCheckResourceNonEmptyList`